src/import-data/ImportData.ts

```diff
@@ -1,72 +1,88 @@
 import { useOnMount } from 'src/libs/react-utils';
 import { asyncImportJobStore } from 'src/libs/state';
 import * as Utils from 'src/libs/utils';
+import { WorkspaceInfo } from 'src/libs/workspace-utils';
 import { useDataCatalog } from 'src/pages/library/dataBrowser-utils';
 import { notifyDataImportProgress } from 'src/workspace-data/import-jobs';
 
-import { TemplateWorkspaceInfo } from './import-types';
+import {
+  BagItImportRequest,
+  CatalogDatasetImportRequest,
+  CatalogSnapshotsImportRequest,
+  EntitiesImportRequest,
+  ImportRequest,
+  PFBImportRequest,
+  TDRSnapshotExportImportRequest,
+  TDRSnapshotReferenceImportRequest,
+  TemplateWorkspaceInfo,
+} from './import-types';
 import { ImportDataDestination } from './ImportDataDestination';
 import { ImportDataOverview } from './ImportDataOverview';
 import { isProtectedSource } from './protected-data-utils';
+import { useImportRequest } from './useImportRequest';
 
-// ImportData handles all the information relating to the page itself - this includes:
-// * Reading from the URL
-// * Loading initial Data
-// * Managing the import
-export const ImportData = () => {
+const getTitleForImportRequest = (importRequest: ImportRequest): string => {
+  if (importRequest.type === 'tdr-snapshot-export') {
+    return `Importing snapshot ${importRequest.snapshotName}`;
+  }
+
+  if (
+    importRequest.type === 'tdr-snapshot-reference' ||
+    importRequest.type === 'catalog-dataset' ||
+    importRequest.type === 'catalog-snapshots'
+  ) {
+    return 'Linking data to a workspace';
+  }
+
+  return 'Importing data to a workspace';
+};
+
+export interface ImportDataProps {
+  importRequest: ImportRequest;
+}
+
+export const ImportData = (props: ImportDataProps): ReactNode => {
+  const { importRequest } = props;
   const {
-    query: {
-      url,
-      format,
-      ad,
-      wid,
-      template,
-      snapshotId,
-      snapshotName,
-      snapshotIds,
-      referrer,
-      tdrmanifest,
-      catalogDatasetId,
-      tdrSyncPermissions,
-    },
+    query: { ad, format, wid, template },
   } = Nav.useRoute();
   const [templateWorkspaces, setTemplateWorkspaces] = useState<{ [key: string]: TemplateWorkspaceInfo[] }>();
   const [userHasBillingProjects, setUserHasBillingProjects] = useState(true);
   const [snapshotResponses, setSnapshotResponses] = useState<{ status: string; message: string | undefined }[]>();
   const [isImporting, setIsImporting] = useState(false);
 
   const { dataCatalog } = useDataCatalog();
-  const snapshots = _.flow(
-    _.filter(
-      (snapshot: Dataset) => !!snapshot['dct:identifier'] && _.includes(snapshot['dct:identifier'], snapshotIds)
-    ),
-    _.map((snapshot) => ({
-      // The previous step filters the list to only datasets with 'dct:identifier' defined
-      id: snapshot['dct:identifier']!,
-      title: snapshot['dct:title'],
-      description: snapshot['dct:description'],
-    }))
-  )(dataCatalog);
+  const snapshots =
```

Comment:
This is only valid for `catalog-snapshots` imports, so only run the flow in that case.